docs/modules/ROOT/pages/api/governance.adoc

```diff
@@ -1982,6 +1990,560 @@ Emits a {TimelockUpdated} event.
+[[IMultisig-get_signers]]
+==== `[.contract-item-name]#++get_signers++#++() → Span<ContractAddress>++` [.item-kind]#external#
+
+Returns the list of all current signers.
+
+[.contract-item]
+[[IMultisig-is_confirmed]]
+==== `[.contract-item-name]#++is_confirmed++#++(id: TransactionID) → bool++` [.item-kind]#external#
+
+Returns whether the transaction with the given `id` has been confirmed.
+
+[.contract-item]
+[[IMultisig-is_confirmed_by]]
+==== `[.contract-item-name]#++is_confirmed_by++#++(id: TransactionID, signer: ContractAddress) → bool++` [.item-kind]#external#
+
+Returns whether the transaction with the given `id` has been confirmed by the specified `signer`.
+
+[.contract-item]
+[[IMultisig-is_executed]]
+==== `[.contract-item-name]#++is_executed++#++(id: TransactionID) → bool++` [.item-kind]#external#
+
+Returns whether the transaction with the given `id` has been executed.
+
+[.contract-item]
+[[IMultisig-get_transaction_state]]
+==== `[.contract-item-name]#++get_transaction_state++#++(id: TransactionID) → TransactionState++` [.item-kind]#external#
+
+Returns the current state of the transaction with the given `id`.
+
+[.contract-item]
+[[IMultisig-get_transaction_confirmations]]
+==== `[.contract-item-name]#++get_transaction_confirmations++#++(id: TransactionID) → u32++` [.item-kind]#external#
+
+Returns the number of confirmations from registered signers for the transaction with the specified `id`.
+
+[.contract-item]
+[[IMultisig-get_submitted_block]]
+==== `[.contract-item-name]#++get_submitted_block++#++(id: TransactionID) → u64++` [.item-kind]#external#
+
+Returns the block number when the transaction with the given `id` was submitted.
+
+[.contract-item]
+[[IMultisig-add_signers]]
+==== `[.contract-item-name]#++add_signers++#++(new_quorum: u32, signers_to_add: Span<ContractAddress>)++` [.item-kind]#external#
+
+Adds new signers and updates the quorum.
+
+Requirements:
+- The caller must be the contract itself.
+- `new_quorum` must be less than or equal to the total number of signers after addition.
+
+Emits a {IMultisig-SignerAdded} event for each signer added.
+Emits a {IMultisig-QuorumUpdated} event if the quorum changes.
+
+[.contract-item]
+[[IMultisig-remove_signers]]
+==== `[.contract-item-name]#++remove_signers++#++(new_quorum: u32, signers_to_remove: Span<ContractAddress>)++` [.item-kind]#external#
+
+Removes signers and updates the quorum.
+
+Requirements:
+- The caller must be the contract itself.
+- `new_quorum` must be less than or equal to the total number of signers after removal.
+
+Emits a {IMultisig-SignerRemoved} event for each signer removed.
+Emits a {IMultisig-QuorumUpdated} event if the quorum changes.
+
+[.contract-item]
+[[IMultisig-replace_signer]]
+==== `[.contract-item-name]#++replace_signer++#++(signer_to_remove: ContractAddress, signer_to_add: ContractAddress)++` [.item-kind]#external#
+
+Replaces an existing signer with a new signer.
+
+Requirements:
+- The caller must be the contract itself.
```

Comment:
On second thought, should we move the requirements to the component implementation descriptions? Since there's no multisig standard, it seems more appropriate to me to have the requirements in the implementation. Perhaps, it should be both, but IMO they def should be in the impl. WDYT?

Reply:
Added to the implementation so that requirements are present in both places to stay consistent with our other docs